modules/core/src/main/java/org/apache/synapse/core/axis2/ProxyService.java

```diff
@@ -448,6 +465,29 @@ public AxisService buildAxisService(SynapseConfiguration synCfg, AxisConfigurati
         // if a WSDL was found
         if (wsdlElement != null) {
             OMNamespace wsdlNamespace = wsdlElement.getNamespace();
+            // if preservePolicy is set to 'false', remove the security policy content of publish wsdl
```

Comment:
buildAxisService method is too long. Better to at least take the new part to a different methode.

Reply:
+1. Moved code to new method.